### PR TITLE
breakup optim, cuda documentation

### PR DESCRIPTION
**Pull request by mattip**
Related to #52256

Use autosummary instead of autofunction to create subpages for optim and cuda functions/classes.

Also fix some minor formatting issues in optim.LBFGS and cuda.stream docstings

### CODE REVIEW

**Comment by mattip** [comment]
@zou3519

File: docs/source/optim.rst [hunk 1]
```diff
@@ -155,26 +161,20 @@ should write your code this way:
...
+    lr_scheduler.LambdaLR
+    lr_scheduler.MultiplicativeLR
+    lr_scheduler.StepLR
+    lr_scheduler.MultiStepLR
+    lr_scheduler.ExponentialLR
+    lr_scheduler.CosineAnnealingLR
+    lr_scheduler.ReduceLROnPlateau
+    lr_scheduler.CyclicLR
+    lr_scheduler.OneCycleLR
+    lr_scheduler.CosineAnnealingWarmRestarts
```
Comment (zou3519) [inline docs/source/optim.rst:177]: This one read a little weirdly:
image

File: docs/source/cuda.rst [hunk 2]
```diff
@@ -3,68 +3,104 @@ torch.cuda
 
 .. currentmodule:: torch.cuda
 
-.. automodule:: torch.cuda
-   :members:
+.. autosummary::
```
Comment (zou3519) [inline docs/source/cuda.rst:6]: Hmm, we lost the nice description:

Before:
image

After:
image

Comment (mattip) [reply docs/source/cuda.rst:6]: restored

**Review by zou3519 — APPROVED** [review]
lgtm!

**Pull request closed without merging**